src/main/java/com/Model/UserDetail.java

```diff
@@ -0,0 +1,64 @@
+package com.Model;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class UserDetail implements UserDetails {
+
+    private String userName;
```

Comment:
aren't these instance variables duplicating what's in the User class? Maybe the User class shouldn't have the same fields?

Reply:
Redundant fields deleted.